android/src/main/java/com/robinpowered/RNAudioJack/RNAudioJackModule.java

```diff
@@ -2,20 +2,83 @@
 
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
 import com.facebook.react.bridge.ReactMethod;
-import com.facebook.react.bridge.Callback;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.WritableNativeMap;
 
-public class RNAudioJackModule extends ReactContextBaseJavaModule {
+import java.util.Map;
+import java.util.HashMap;
+import javax.annotation.Nullable;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Build;
+import android.media.AudioDeviceInfo;
+import android.media.AudioManager;
 
-  private final ReactApplicationContext reactContext;
+public class RNAudioJackModule extends ReactContextBaseJavaModule {
+  private static final String MODULE_NAME = "RNAudioJack";
```

Comment:
Little backstory on why you generally see `RN` in the prefix here: the origins come from iOS.

In iOS, files (classes) are not namespaced (unless you use frameworks, which isn't common). So you'll usually see al of `RCT*` or `RN*` prefixes. 

Again in iOS, when you create the module, you call a macro `RCT_EXPORT_MODULE()`, which creates a native module with the same name as the class name. And since the class name will use some sort of package identifier in its name ("RN"), it becomes the module name, and then you'd end up doing the same on Android to keep it consistent with the name generated by iOS.

HOWEVER -- you can now name your module in iOS with `RCT_EXPORT_MODULE("NAME")`.

----

Given that this is an open source project, we want to make sure the files are namespaced. So the iOS filename/class should continue to be `RNAudioJack`, but we can use `RCT_EXPORT_MODULE("AudioJack")` to create a friendly name for the module.

TLDR - remove it, but it was put there with good intentions